src/Gambot.Core/Message.cs

```diff
@@ -10,11 +12,13 @@ public class Message
         public string Channel { get; }
         public Person From { get; }
         public string To { get; }
+        public IDictionary<string, string> Variables { get; }

```

Comment:
Did you add this just to enable passing pre-populated variable replacements to the factoid responder?  That's ... impressive.

I'm not sure I want the core message to be aware of variables, though.  Like I said on the responder itself, this feels a bit overengineered.

Having said that, I wonder if there's something we could introduce to abstract the useful bits of the factoid selection logic since this is becoming a common use case.

Reply:
This did feel pretty sloppy. It was necessary because of how I handled the factoid piece, the InventoryResponder couldn't just string-replace "$item" because it was getting the response from the FactoidResponder, and the text on a Response is immutable. (Also, string-replace on $item wouldn't have handled capitalization as well).

I'm open to alternatives. If Gambot.Core exposed an IFactoidResolver and IVariableReplacer, this new module wouldn't need to have so many tendrils into other modules.